### ISSUE TITLE
Add rule to suggest `xargs -I{}` over `xargs -i`

### ISSUE THREAD

**Issue by nevesnunes** [koalaman/shellcheck#2058]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
printf '%s\n' 1 2 | xargs -i echo {}
```

#### Here's what shellcheck currently says:

N/A

#### Here's what I wanted or expected to see:

Some xargs don't have option `-i`. Use `-I{}` instead.

#### Context

- BSD implementations do not support `-i`. Reference: https://www.freebsd.org/cgi/man.cgi?query=xargs
- GNU implementation deprecates `-i`. Reference: https://git.savannah.gnu.org/cgit/findutils.git/commit/?id=0b1f485a14102dfa58ec4eb66260f70e26496430

**Comment by koalaman** [comment]
You're right. There's now a warning about this, thanks!
